### DISCUSSION TITLE
How make toast hide after few second without button

### DISCUSSION THREAD

**Discussion by Laszlo7801** [twbs/bootstrap#40800] [Q&A] [answered]
Hi,
As other people, I'm experiencing difficulties to make toast hide after few second. I've found this code that works : https://codepen.io/louismaximepiton/pen/xxmGLQj

However, now I'm looking for an adaptation of this code to show the toast when the page is loaded and make the toast disapear automatticaly after few seconds.

Could you help me ?

**Comment by ANU1o** [comment, marked as answer]
try this

### HTML
`<body onload="loaderToast()"></body>`
> **Note:** If it is raw HTML use this. Otherwise while using any framework, use the outermost tag or `onload` function related to the framework

### Javascript
`function loaderToast () {
  const toastBootstrap = bootstrap.Toast.getOrCreateInstance(toastLiveExample)
  toastBootstrap.show()
}`

**Reply by Laszlo7801** [reply]
Works like a charm.

Thanks ANU1o !

**Comment by mrrizwan304** [comment]
To make a toast hide automatically after a few seconds without needing a button, you can use JavaScript to set a timer. Here's an example using HTML, CSS, and JavaScript:

HTML:
html
Copy code
<div id="toast" class="toast">This is a toast message!</div>
CSS:
css
Copy code
.toast {
    visibility: hidden;
    min-width: 250px;
    background-color: #333;
    color: white;
    text-align: center;
    padding: 16px;
    position: fixed;
    z-index: 1;
    left: 50%;
    bottom: 30px;
    transform: translateX(-50%);
    border-radius: 4px;
}

.toast.show {
    visibility: visible;
    animation: fadeOut 0.5s 5s forwards; /* Fades out after 5 seconds */
}

@keyframes fadeOut {
    0% { opacity: 1; }
    100% { opacity: 0; }
}
JavaScript:
javascript
Copy code
function showToast() {
    var toast = document.getElementById("toast");
    toast.classList.add("show");
    
    // Hide the toast after 5 seconds
    setTimeout(function(){
        toast.classList.remove("show");
    }, 5000); // 5000ms = 5 seconds
}

// Call the function to display the toast
showToast();
Explanation:
The .show class is added to make the toast visible.
setTimeout is used to automatically remove the .show class after 5 seconds, causing the toast to disappear.
The @keyframes fadeOut is used to animate the fade-out effect.
When showToast() is called, the toast will appear and automatically hide after 5 seconds.

**Reply by ANU1o** [reply]
In his source code `data-bs-delay="1000"` is already doing that. He'll need to adjust it for disappearing.

> Note that 1000 means 1 second.
